### PR TITLE
NIJO-1110. Fix MemoryCache issues

### PR DESCRIPTION
**Pull request by oleg1std**
Add MemoryCache in the services and fix getting an access token from the cache

### CODE REVIEW

File: HappyTravel.Edo.BookingStatusUpdater/Infrastructure/IdentityServerClient.cs [hunk 1]
```diff
@@ -20,14 +20,16 @@ public class IdentityServerClient : IIdentityServerClient
 
         public async Task<string> RequestClientCredentialsTokenAsync()
         {
-            if (!_cache.TryGetValue<string>("accessToken", out var accessToken))
+            const string accessTokenKey = "accessToken";
```
Comment (DFSko) [inline HappyTravel.Edo.BookingStatusUpdater/Infrastructure/IdentityServerClient.cs:23]: Maybe replace this part and get access token like in travelgate channel?

Comment (mingazhev) [reply HappyTravel.Edo.BookingStatusUpdater/Infrastructure/IdentityServerClient.cs:23]: And in EDO, there is a ready and fine-working approach

**Review by mingazhev — APPROVED** [review]

File: HappyTravel.Edo.BookingStatusUpdater/Infrastructure/IdentityServerClient.cs (hunk 1 above)
Comment (oleg1std) [reply HappyTravel.Edo.BookingStatusUpdater/Infrastructure/IdentityServerClient.cs]: Ok. This part has been changed

File: HappyTravel.Edo.BookingStatusUpdater/Infrastructure/EnvironmentVariableHelper.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+
+namespace HappyTravel.Edo.BookingStatusUpdater.Infrastructure
+{
+    namespace HappyTravel.TravelgateXChannel.Infrastructure
```
Comment (DFSko) [inline HappyTravel.Edo.BookingStatusUpdater/Infrastructure/EnvironmentVariableHelper.cs]: Namespace

Comment (oleg1std) [reply HappyTravel.Edo.BookingStatusUpdater/Infrastructure/EnvironmentVariableHelper.cs]: It was removed

Comment (oleg1std) [reply HappyTravel.Edo.BookingStatusUpdater/Infrastructure/EnvironmentVariableHelper.cs]: Helper, I mean

**Review by DFSko — APPROVED** [review]

File: HappyTravel.Edo.BookingStatusUpdater/Startup.cs [hunk 3]
```diff
@@ -28,38 +30,10 @@ public void ConfigureServices(IServiceCollection services)
...
-            {
-                client.BaseAddress = new Uri(edoApiUrl);
-                client.DefaultRequestHeaders.Add("Accept", "application/json");
-            })
-            .AddHttpMessageHandler<ProtectedApiBearerTokenHandler>();
-            
+            services.AddHttpClients(_configuration, vaultClient);
             services.AddHostedService<StatusUpdateService>();
             services.AddHealthChecks();
+            services.AddMemoryCache();
```
Comment (DFSko) [inline HappyTravel.Edo.BookingStatusUpdater/Startup.cs:36]: Not needed now

**Pull request merged**